public/video-ui/src/slices/s3Upload.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice,PayloadAction } from '@reduxjs/toolkit';
+
+export interface S3UploadState {
+  id: string | null;
+  progress: number;
+  total: number;
+  status: 'idle' | 'uploading' | 'post-processing' | 'complete' | 'error';
+}
+
+type S3UploadAction = PayloadAction<S3UploadState>
+
+const initialState : S3UploadState =  { id: null, progress: 0, total: 0, status: 'idle' };
+
+const s3Upload = createSlice({
+  name: 's3Upload',
+  initialState,
+  reducers: {
+    s3UploadStarted:(state :S3UploadState, action)=> {
+      const total = action.payload.parts[action.payload.parts.length - 1].end;
+      Object.assign(state, {
+        id: action.payload.id,
+        total: total,
+        status: 'uploading'
+      });
+    },
+    s3UploadProgress: (state : S3UploadState, action: S3UploadAction)=>{
+      Object.assign(state, {
+        progress: action.payload.progress,
```

Comment:
As noted above - the payload will be a number, so setting `progress` to `action.payload.progress` will be setting it to `undefined`.
```suggestion
        progress: action.payload,
```

Reply:
it would be best to not use `Object.assign` as we loose type safety that way - I think it was the norm in the js reducer files as it meant they didn't need to track which fields where being changed that way.